ra/ra.go

```diff
@@ -1375,10 +1375,14 @@ func (ra *RegistrationAuthorityImpl) checkCertificatesPerFQDNSetLimit(ctx contex
 	}
 	names = core.UniqueLowerNames(names)
 	threshold := limit.GetThreshold(strings.Join(names, ","), regID)
-	if count.Count >= threshold {
+	if int64(len(issuanceTimestampsInWindow.Timestamps)) >= threshold {
+
+		// FQDNSetTimestampsForWindow returns the issuance timestamps in
+		// ascending order so we can assume that the first one is the oldest.
+		retryAfter := time.Unix(0, issuanceTimestampsInWindow.Timestamps[0]).Add(limit.Window.Duration)
```

Comment:
There's a chance that `threshold` is 0, we enter this block, and then panic when we try to index into the empty `Timestamps` slice.

Unrelated: also put a TODO(#6154) here, so that we remember to update this message when we update the way the ratelimit is calculated.

Reply:
> There's a chance that threshold is 0, we enter this block, and then panic when we try to index into the empty Timestamps slice.

Ah, actually it's a little worse than that. It completely slipped my mind, but we actually need to check for a `databaseOpErr`, specifically `NoRows` here. The previous method used a count query which would return `0`, but I believe this one would return a `NoRows` error.

Reply:
That's a good catch. But I would expect it to be caught by unit / integration tests since the first issuance for any FQDNSet would hit that path. Any idea why that didn't happen?

Reply:
Ah, valid point! I'll look into that.

Reply:
My bad, I mixed up the possible values for `Select()` vs `SelectOne()`.

Reply:
I think the fix for this particular issue is that we shouldn't calculate `issuanceTimestampsInWindow.Timestamps[0]` unless `len(issuanceTimestampsInWindow.Timestamps) > 0`. Above, we check that:

```
int64(len(issuanceTimestampsInWindow.Timestamps)) >= threshold
```

Perhaps we can also check for `threshold <= 0`, above that, and early-return from the function with a nil error?